core/tests/puppeteer-acceptance-tests/utilities/common/puppeteer-utils.ts

```diff
@@ -759,6 +771,49 @@ export class BaseUser {
     this.page = newPage;
     return newPage;
   }
+
+  /**
+   * This function sets up video recording for the current test.
+   */
+  async setupVideoRecording(): Promise<void> {
+    const outputFileName =
+      `${this.specName}-${new Date().toISOString()}.mp4`.replace(
+        /[^a-z0-9.-]/gi,
+        '_'
+      );
+
+    const outputDir = testConstants.TEST_VIDEO_DIR;
+    if (!fs.existsSync(outputDir)) {
+      fs.mkdirSync(outputDir, {recursive: true});
+    }
+
+    // Skip if there is no page object.
+    if (!this.page) {
+      return;
+    }
+
+    const config = {
+      followNewTab: true,
+      fps: 25,
+      ffmpeg_Path: null,
+      videoFrame: {
+        width: this.page.viewport()?.width || 1920,
```

Comment:
Not sure about hardcoding 1920/1080 here. What about mobile viewports?

Do we even need to hardcode it?

Reply:
When using mobile (or even desktop) viewport is already set in `openBrowser`. Same viewport will be used except for case when there is no viewport (in which 1920/1080) will be used.

Reply:
Why would there be no viewport?

Also I think this context you're giving is useful. It is probably a good idea to include that as a comment in the file explaining why we are doing what we're doing. (In general, when a reviewer leaves comments, consider adding comments in the code to clarify as needed.)

Reply:
Got It. I will add comments to code when reviewer asks questions. Thanks @seanlip!

Again, this was proofing against developers using this video recording function without configuring viewport of page correctly. However, I checked documentation and it says viewport is set to some default values when viewport isn't passed. So, we are not required null check (`?`).

Will remove it.